### DISCUSSION TITLE
rm -rf protobuf

### DISCUSSION THREAD

**Discussion by wilderfield** [microsoft/onnxruntime#20655] [Ideas / Feature Requests]
Hello,

I am writing here to promote the anarchistic view of removing dependence on protobuf.

Rationale:
- Protobuf has a file size limit of 2GB. Cutting edge models today are larger than that, and dealing with models that have external weights is cumbersome. Several projects like llama.cpp define their own model format that doesn't have this issue.
- Many libraries use protobuf, and protobuf releases do not play well with each other. Many-a-dependency hell has been attributed to resolving multiple versions of protobuf in an application.

Why not use your own serialization library?

**Comment by Craigacp** [comment]
The ONNX format is defined as a protobuf - https://github.com/onnx/onnx/blob/main/docs/IR.md.
